### PR TITLE
#159732732 Enable CRUD operations on a player.

### PR DESCRIPTION
**Pull request by kaguna**
#### What does this PR do?
- Enables a user to perform CRUD operations on a player.

#### Description of Task to be completed?
- Create a controller file to perform CRUD operations.
- Write a faker for player details.
- Test the CRUD operations

#### How should this be manually tested?
- 
#### Pivotal tracker story
- [#159732732](https://www.pivotaltracker.com/story/show/159732732)

### CODE REVIEW

File: spec/factories/players.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FactoryBot.define do
+
+  factory :player, class: Player do
+    first_name { Faker::Name.name }
```
Comment (gr1d99) [inline spec/factories/players.rb]: This can be refactored to `Faker::Name.first_name`

File: spec/factories/players.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    second_name { Faker::Name.name }
```
Comment (gr1d99) [inline spec/factories/players.rb]: This can also be refactored to `Faker::Name.last_name`

File: spec/requests/players_spec.rb [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        first_name: "James",
+        second_name: "Kariuki",
+        last_name: "Kaguna",
+        nick_name: "jimnah",
+        id_number: "31262778",
+        dob: "1997-09-19",
+        phone_number: "0715739940",
+        team_id: team_id
+    }
+  }
```
Comment (gr1d99) [inline spec/requests/players_spec.rb]: Did you try using **FactoryBot** methods?
- it has an `attributes_for` method which achieves what this block is doing. 
ie 
```ruby
...
let(:player_params) { attributes_for(player.first) }
...
```

File: spec/requests/players_spec.rb [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            first_name: "",
+            second_name: "Kariuki",
+            last_name: "Kaguna",
+            nick_name: "jimnah",
+            id_number: "31262778",
+            dob: "1997-09-19",
+            phone_number: "0715739940",
+            team_id: player_id
+        }
+      }
```
Comment (gr1d99) [inline spec/requests/players_spec.rb]: the same suggestion in line **24** applies

File: spec/requests/players_spec.rb [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+      before { post "/players/#{team_id}", params: player_params }
+
+      it "does not create a new player with empty player first name" do
+        expect(json["first_name"]).to eq(["can't be blank"])
+      end
+
+      it "returns status code 422" do
+        expect(response).to have_http_status(422)
+      end
```
Comment (gr1d99) [inline spec/requests/players_spec.rb]: we can use `shoulda` matchers here to reduce this three lines into **one**!,
`it { is_expected.to respond_with(422) }`

File: spec/requests/players_spec.rb [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    context "when the request is valid" do
+      before { get "/players/#{team_id}/#{player_id}" }
+
+      it "returns a hash of one player with 11 elements" do
+        expect(json.size).to eq 11
+      end
+
+      it "returns status code 200" do
+        expect(response).to have_http_status(200)
+      end
```
Comment (gr1d99) [inline spec/requests/players_spec.rb]: `shoulda` matchers also needed here

File: spec/requests/players_spec.rb [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+    end
+
+    context "when the request is invalid" do
+      let(:team_id) { 0 }
+      before { get "/players/#{team_id}/#{player_id}" }
+
+      it "returns an error message and status code 400" do
+        expect(response).to have_http_status(400)
+        expect(json["errors"]).to eq("The player does not exist")
```
Comment (gr1d99) [inline spec/requests/players_spec.rb]: let's make each example test one thing at a time, one example for status code and the other for expected message.

**Comment by kaguna** [comment]
@gr1d99 changes have made. However, using `shoulda matcher` does not seem to work from my end.

**Pull request merged**